snapshots/btrfs/btrfs.go

```diff
@@ -194,47 +183,28 @@ func (b *snapshotter) View(ctx context.Context, key, parent string, opts ...snap
 	return b.makeSnapshot(ctx, snapshots.KindView, key, parent, opts)
 }
 
-func (b *snapshotter) makeSnapshot(ctx context.Context, kind snapshots.Kind, key, parent string, opts []snapshots.Opt) ([]mount.Mount, error) {
-	ctx, t, err := b.ms.TransactionContext(ctx, true)
-	if err != nil {
-		return nil, err
-	}
-	defer func() {
-		if err != nil && t != nil {
-			if rerr := t.Rollback(); rerr != nil {
-				log.G(ctx).WithError(rerr).Warn("failed to rollback transaction")
-			}
-		}
-	}()
+func (b *snapshotter) makeSnapshot(ctx context.Context, kind snapshots.Kind, key, parent string, opts []snapshots.Opt) (_ []mount.Mount, err error) {
+	target, s := "", storage.Snapshot{}
 
-	s, err := storage.CreateSnapshot(ctx, kind, key, parent, opts...)
-	if err != nil {
-		return nil, err
-	}
+	err = b.ms.WithTransaction(ctx, true, func(ctx context.Context) error {
+		s, err = storage.CreateSnapshot(ctx, kind, key, parent, opts...)
+		if err != nil {
+			return err
+		}
 
-	target := filepath.Join(b.root, strings.ToLower(s.Kind.String()), s.ID)
+		target = filepath.Join(b.root, strings.ToLower(s.Kind.String()), s.ID)
 
-	if len(s.ParentIDs) == 0 {
-		// create new subvolume
-		// btrfs subvolume create /dir
-		if err = btrfs.SubvolCreate(target); err != nil {
-			return nil, err
+		if len(s.ParentIDs) == 0 {
+			// create new subvolume
+			// btrfs subvolume create /dir
+			return btrfs.SubvolCreate(target)
 		}
-	} else {
 		parentp := filepath.Join(b.root, "snapshots", s.ParentIDs[0])
 
-		var readonly bool
-		if kind == snapshots.KindView {
-			readonly = true
-		}
-
 		// btrfs subvolume snapshot /parent /subvol
-		if err = btrfs.SubvolSnapshot(target, parentp, readonly); err != nil {
-			return nil, err
-		}
-	}
-	err = t.Commit()
-	t = nil
+		return btrfs.SubvolSnapshot(target, parentp, kind == snapshots.KindView)
```

Comment:
Kind of a nit, but the explicit readOnly assignment made this slightly more readable. I'd have to go look at `SubVolSnapshot`'s declaration to see what's the significance of either true or false (or be versed in what KindView signifies in containerd parlance), but before I can take a quick glance and go "oh, it's likely for a read only snapshot".